chef/cookbooks/provisioner/libraries/repositories.rb

```diff
@@ -52,24 +62,40 @@ def inspect_repos(node)
         case node[:platform]
         when "suse"
           repos = Mash.new
-          missing = false
+          missing_hae = false
+          missing_storage = false
+
           %w(11.3 12.0).each do |version|
             repos.merge! suse_get_repos_from_attributes(node,"suse",version)
 
-            suse_optional_repos(version).each do |name|
+            # For pacemaker
+            suse_optional_repos(version, :hae).each do |name|
+              repos[name] ||= Mash.new
+              next unless repos[name][:url].nil?
+              missing_hae ||= !(File.exists? "#{node[:provisioner][:root]}/repos/#{name}/repodata/repomd.xml")
+            end
+
+            # For suse storage
+            suse_optional_repos(version, :storage).each do |name|
               repos[name] ||= Mash.new
               next unless repos[name][:url].nil?
-              missing ||= !(File.exists? "#{node[:provisioner][:root]}/repos/#{name}/repodata/repomd.xml")
+              missing_storage ||= !(File.exists? "#{node[:provisioner][:root]}/repos/#{name}/repodata/repomd.xml")
             end
           end
 
           # set an attribute about missing repos so that cookbooks and crowbar
           # know that HA cannot be used
+          # know that SUSE_Storage cannot be used
           node.set[:provisioner][:suse] ||= {}
-          if node[:provisioner][:suse][:missing_hae] != missing
-            node.set[:provisioner][:suse][:missing_hae] = missing
-            node.save
+          if node[:provisioner][:suse][:missing_hae] != missing_hae
+            node.set[:provisioner][:suse][:missing_hae] = missing_hae
+          end
+          if node[:provisioner][:suse][:missing_storage] != missing_storage
+            node.set[:provisioner][:suse][:missing_storage] = missing_storage
           end
+
+          node.save
```

Comment:
Can you save only if there was a `node.set`? (Just use a variable to know if there's a need for saving)